views/decrypt.ejs

```diff
@@ -31,11 +31,20 @@
                     <input type="password" class="form-control" name="key">
                     <small class="form-text text-muted">Only individuals with this key can decrypt your URL.</small>
                 </div>
+                <% if (locals.decrypt_fail) { %>
+                    <script>
+                      let encryptedUrl = '<%- encryptedUrl %>';
+                      window.history.pushState("", "", '/' + encryptedUrl);
```

Comment:
This is the only way I found to have the address bar keep the encryptedURL instead of losing it when the decrypt fails. There may very well also be a better way.